File: back/src/main/java/com/dentura/api/appointment/Appointment.java
package com.dentura.api.appointment;

import java.time.Instant;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;

@Entity
@Table(name = "appointments")
public class Appointment {

	public static final String SCHEDULED = "SCHEDULED";
	public static final String CONFIRMED = "CONFIRMED";
	public static final String CANCELLED = "CANCELLED";
	public static final String COMPLETED = "COMPLETED";
	public static final String NO_SHOW = "NO_SHOW";

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(name = "clinic_id", nullable = false)
	private Long clinicId;

	@Column(name = "patient_id", nullable = false)
	private Long patientId;

	@Column(name = "start_at", nullable = false)
	private Instant startAt;

	@Column(name = "end_at", nullable = false)
	private Instant endAt;

	@Column(nullable = false, length = 20)
	private String status = SCHEDULED;

	@Column(length = 255)
	private String reason;

	@Column(columnDefinition = "TEXT")
	private String notes;

	@Column(name = "created_at", nullable = false)
	private Instant createdAt;

	@Column(name = "updated_at", nullable = false)
	private Instant updatedAt;

	@PrePersist
	void onCreate() {
		Instant now = Instant.now();
		createdAt = now;
		updatedAt = now;
	}

	@PreUpdate
	void onUpdate() {
		updatedAt = Instant.now();
	}

	public Long getId() {
		return id;
	}

	public Long getClinicId() {
		return clinicId;
	}

	public void setClinicId(Long clinicId) {
		this.clinicId = clinicId;
	}

	public Long getPatientId() {
		return patientId;
	}

	public void setPatientId(Long patientId) {
		this.patientId = patientId;
	}

	public Instant getStartAt() {
		return startAt;
	}

	public void setStartAt(Instant startAt) {
		this.startAt = startAt;
	}

	public Instant getEndAt() {
		return endAt;
	}

	public void setEndAt(Instant endAt) {
		this.endAt = endAt;
	}

	public String getStatus() {
		return status;
	}

	public void setStatus(String status) {
		this.status = status;
	}

	public String getReason() {
		return reason;
	}

	public void setReason(String reason) {
		this.reason = reason;
	}

	public String getNotes() {
		return notes;
	}

	public void setNotes(String notes) {
		this.notes = notes;
	}

	public Instant getCreatedAt() {
		return createdAt;
	}

	public Instant getUpdatedAt() {
		return updatedAt;
	}
}

File: front/src/@types/appointment.ts
export type AppointmentStatus =
    | 'SCHEDULED'
    | 'CONFIRMED'
    | 'CANCELLED'
    | 'COMPLETED'
    | 'NO_SHOW'

export type Appointment = {
    id: number
    clinicId: number
    patientId: number
    patientName: string
    recordNumber: string
    providerId: number
    providerName: string
    providerColor: string
    roomId: number | null
    roomName: string | null
    startAt: string
    endAt: string
    status: AppointmentStatus
    reason: string | null
    notes: string | null
    createdAt: string
    updatedAt: string
}

export type AppointmentPayload = {
    patientId: number
    providerId?: number | null
    roomId?: number | null
    startAt: string
    endAt: string
    status?: AppointmentStatus
    reason?: string | null
    notes?: string | null
}

export type Provider = {
    id: number
    clinicId: number
    name: string
    color: string
    userId: number | null
    active: boolean
}

export type Room = {
    id: number
    clinicId: number
    name: string
    active: boolean
}
